pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -2665,7 +2917,13 @@ func (p *Provider) List(l *logger.Logger, opts vm.ListOptions) (vm.List, error)
 				// The former is a subset of the latter. Some information like `Labels` will be missing.
 				disks = toDescribeVolumeCommandResponse(jsonVM.Disks, jsonVM.Zone)
 			}
-			vms = append(vms, *jsonVM.toVM(prj, disks, defaultOpts, p.publicDomain))
+			v := jsonVM.toVM(prj, disks, defaultOpts, p.publicDomain)
+			vmArch, err := p.GetVMArchitecture(l, v)
+			if err != nil {
+				return nil, errors.Wrap(err, "unable to determine the VM CPU architecture")
```

Comment:
If we bail out here, the entire listing is effectively aborted. Note that occasionally GCP can fail to return cpu arch for a specific instance, or there could be a transient API error. Bailing out is disruptive if only because `roachtest` may fail [1]. Instead, we should probably treat this as a warning, and live with the fact that in a minuscule number of cases, the cpu architecture may not be available.

[1] https://github.com/cockroachdb/cockroach/issues/130082